internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -508,12 +498,15 @@ func (r *PreprovisioningImageReconciler) getIronicConfig(ctx context.Context, lo
 	}
 
 	if iccConfig.IronicBaseURL == "" && iccConfig.IronicInspectorBaseUrl == "" {
-		err := r.fillIronicServiceURLs(ctx, infraEnv, infraEnvInternal, iccConfig)
-		if err != nil {
+		if err := r.fillIronicServiceURLs(ctx, infraEnv, infraEnvInternal, iccConfig); err != nil {
 			return nil, err
 		}
 	}
 
+	if ironicAgentImageUserOverride := r.getIronicAgentImageFromUserOverride(log, infraEnv); ironicAgentImageUserOverride != "" {
```

Comment:
I think this needs to be earlier so we don't get into a situation where if the icc CM is missing, and all the other image fetching fails. Currently we would get the "Failed to determine ironic config" error, but we should be allowed to use the override.